eval.c

```diff
@@ -259,9 +260,10 @@ static sexp sexp_flatten_dot (sexp ctx, sexp ls) {
 
 int sexp_param_index (sexp ctx, sexp lambda, sexp name) {
   sexp ls;
-  int i;
+  int i = 0;
   while (1) {
-    i = 0;
+    // ISSUE_296 - Value stored to 'i' is never used
+    //i = 0;
```

Comment:
This is a pretty useless static analyzer warning.  The value of i is initialized upon the first pass of the for loop, which overrides the initialization value the analyzer is complaining about a few lines before.  It's just redundant code.  Initializing outside the loop also performs no function (it would be redundant) as the first thing the for loop does is set i to 0; but initializing things in the declaration is often considered good practice should never break anything; otherwise the logic would be depending on the initialized value.

Removed the redundant i = 0 statement to get rid of the warning.
```c
 while (1) {
    ls = sexp_lambda_params(lambda);
    for (i=0; sexp_pairp(ls); ls=sexp_cdr(ls), i++)
      if (sexp_car(ls) == name)
        return i;
...
```